syntax = "proto3";

message Text {
  string data = 1;
}

message Image {
  bytes img = 1;
}

service DataHash {
  rpc get_image(Text) returns (Image) {};
  rpc hash_md5(Text) returns (Text) {};
  rpc hash_sha256(Text) returns (Text) {};
  rpc hello(Text) returns (Text) {};
  rpc stream_text(Text) returns (stream Text) {}
  rpc input_stream(stream Text) returns (Text) {}
  rpc dual_stream(stream Text) returns (stream Text) {}
}
